test/integration/scheduler/queueing/queue_test.go

```diff
@@ -641,26 +641,95 @@ func TestCoreResourceEnqueue(t *testing.T) {
+			initialPods: []*v1.Pod{
+				st.MakePod().Name("pod1").Label("anti1", "anti1").Container("image").Node("fake-node").Obj(),
+			},
+			pods: []*v1.Pod{
+				// - pod2, pod3, pod4 will be rejected by the PodAffinity plugin.
+				st.MakePod().Name("pod2").Label("anti1", "anti1").PodAntiAffinityExists("anti1", "node", st.PodAntiAffinityWithRequiredReq).Container("image").Obj(),
+				st.MakePod().Name("pod3").Label("anti1", "anti1").PodAntiAffinityExists("anti1", "service", st.PodAntiAffinityWithRequiredReq).Container("image").Obj(),
+				st.MakePod().Name("pod4").Label("anti1", "anti1").PodAntiAffinityExists("anti1", "other", st.PodAntiAffinityWithRequiredReq).Container("image").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
+				// Remove "node" label from the node which will make it not match pod2's podAntiAffinity.
+				// Change "service" label from service-a to service-b which may pod3 schedulable.
+				if _, err := testCtx.ClientSet.CoreV1().Nodes().Update(testCtx.Ctx, st.MakeNode().Name("fake-node").Label("service", "service-b").Label("region", "fake-node").Label("other", "fake-node").Obj(), metav1.UpdateOptions{}); err != nil {
+					return nil, fmt.Errorf("failed to update fake-node: %w", err)
+				}
+				return map[framework.ClusterEvent]uint64{{Resource: framework.Node, ActionType: framework.UpdateNodeLabel}: 1}, nil
+			},
+			wantRequeuedPods:          sets.New("pod2", "pod3"),
+			enableSchedulingQueueHint: []bool{true},
+		},
+		{
+			name:         "Pod rejected by the interPodAffinity plugin is requeued when creating the node with the topologyKey label of the pod anti affinity",
+			initialNodes: []*v1.Node{st.MakeNode().Name("fake-node1").Label("zone", "fake-node").Label("node", "fake-node").Obj()},
+			initialPods: []*v1.Pod{
+				st.MakePod().Name("pod1").Label("anti1", "anti1").Container("image").Node("fake-node1").Obj(),
+			},
+			pods: []*v1.Pod{
+				// - pod2, pod3 will be rejected by the PodAntiAffinity plugin.
+				st.MakePod().Name("pod2").PodAntiAffinityExists("anti1", "zone", st.PodAntiAffinityWithRequiredReq).Container("image").Obj(),
+				st.MakePod().Name("pod3").PodAntiAffinityExists("anti1", "node", st.PodAntiAffinityWithRequiredReq).Container("image").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) (map[framework.ClusterEvent]uint64, error) {
+				// Create the node which will make pod2, pod3 be schedulable.
+				if _, err := testCtx.ClientSet.CoreV1().Nodes().Create(testCtx.Ctx, st.MakeNode().Name("fake-node2").Label("zone", "fake-node").Obj(), metav1.CreateOptions{}); err != nil {
+					return nil, fmt.Errorf("failed to create fake-node2: %w", err)
+				}
+				return map[framework.ClusterEvent]uint64{{Resource: framework.Node, ActionType: framework.Add}: 1}, nil
+			},
+			wantRequeuedPods:          sets.New("pod2", "pod3"),
+			enableSchedulingQueueHint: []bool{true},
```

Comment:
I think here you can test with hints both enabled and disabled

Reply:
done